### PR TITLE
Api extractor fix

### PR DESCRIPTION
**Pull request by kkjeer**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

1. Modify the api-extractor script to use the Extractor object and throw an error if api-extractor fails
2. Modify the update-api script to run this workflow:
a. Run the ts build task so api-extractor picks up the latest code changes
b. Run the api-extractor build task to check whether the api.ts file is up to date
c. If the api.ts file is not up to date, run the api-extractor task with a --local argument to update the api.ts file
d. Re-run the api-extractor task to verify that the updated api.ts file passes the API Extractor requirements

#### Focus areas to test

(optional)

### CODE REVIEW

File: common/changes/@uifabric/set-version/api-extractor-fix_2018-11-01-18-43.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@uifabric/set-version",
+      "comment": "Disable api-extractor build task",
+      "type": "patch"
```
Comment (cliffkoh) [inline common/changes/@uifabric/set-version/api-extractor-fix_2018-11-01-18-43.json]: Build task edits should be none

File: scripts/update-api.js [hunk 2]
```diff
@@ -1,3 +1,51 @@
...
+require('./tasks/ts')({})
+  .then(() => {
+    console.log(cyan, `- Update API: compiled successfully, checking ${apiFileName}...`);
+    checkApi();
+  })
+  .catch(ex => {
+    console.error(red, 'Update API: failed to compile.');
+  });
+
+// Checks the api.ts file to see whether it needs to be updated.
```
Comment (cliffkoh) [inline scripts/update-api.js]: nit: change these to jsdoc format.

File: common/changes/@uifabric/set-version/api-extractor-fix_2018-11-01-18-43.json (hunk 1 above)
Comment (kkjeer) [reply common/changes/@uifabric/set-version/api-extractor-fix_2018-11-01-18-43.json]: This was due to the change made in package.json to add api-extractor to the list of disabled tasks.

Comment (cliffkoh) [reply common/changes/@uifabric/set-version/api-extractor-fix_2018-11-01-18-43.json]: Still should be none though. We don't need to publish for a change like this.

**Comment by KevinTCoughlin** [comment]
This worked great in my testing.

**My Test Flow:**

- Changed `DetailsList.types.ts` file by adding an optional prop.
- Ran `npm run update-api` in the OUIFR project folder since I changed an interface within OUIFR
- The `update-api` script built the project with my changes, generated an up-to-date `d.ts` file, and replaced the existing file.
- It then _prompted_ me to _commit_ the updated API file.

**Output**

```shell
~\office-ui-fabric-react\packages\office-ui-fabric-react [(fbe1066...)]> npm run update-api

> office-ui-fabric-react@6.97.1 update-api ~\office-ui-fabric-react\packages\office-ui-fabric-react
> node ../../scripts/update-api.js

- TS: 'lib' done!
- TS: 'lib-commonjs' done!
- Update API: compiled successfully, checking etc/office-ui-fabric-react.api.ts...
(Verbose) Writing: ~\office-ui-fabric-react\packages\office-ui-fabric-react\dist\office-ui-fabric-react.api.json
You have changed the public API signature for this project. Please overwrite etc/office-ui-fabric-react.api.ts with a copy of temp/office-ui-fabric-react.api.ts and then request an API review. See the Git repository README.md for more info.
- Update API: etc/office-ui-fabric-react.api.ts is out of date, updating...
(Verbose) Writing:~\office-ui-fabric-react\packages\office-ui-fabric-react\dist\office-ui-fabric-react.api.json
You have changed the public API signature for this project. Updating etc/office-ui-fabric-react.api.ts
- Update API: successfully updated etc/office-ui-fabric-react.api.ts, verifying the updates...
(Verbose) Writing: ~\office-ui-fabric-react\packages\office-ui-fabric-react\dist\office-ui-fabric-react.api.json
(Verbose) The API signature is up to date: temp/office-ui-fabric-react.api.ts
- Update API: successfully verified etc/office-ui-fabric-react.api.ts. Please commit etc/office-ui-fabric-react.api.ts as part of your changes.
```

Thank you for these changes 👍.

**Comment by cliffkoh** [comment]
Thank you Kevin for such a detailed testing.

**Pull request merged**